system-addon/test/unit/lib/UserDomainAffinityProvider.test.js

```diff
@@ -30,27 +30,20 @@ const PARAMETER_SETS = {
 };
 
 describe("User Domain Affinity Provider", () => {
-  let UserDomainAffinityProvider;
   let instance;
   let globals;
 
   beforeEach(() => {
     globals = new GlobalOverrider();
-    globals.set("Services", {io: {newURI: u => ({host: "www.somedomain.org"})}});
-    globals.set("PlacesUtils", {
-      history: {
-        getNewQuery: () => ({"TIME_RELATIVE_NOW": 1}),
-        getNewQueryOptions: () => ({}),
-        executeQuery: () => ({root: {childCount: 1, getChild: index => ({uri: "www.somedomain.org", accessCount: 1})}})
-      }
-    });
-    global.Cc["@mozilla.org/browser/nav-history-service;1"] = {
-      getService() {
-        return global.PlacesUtils.history;
-      }
-    };
 
-    ({UserDomainAffinityProvider} = injector());
+    const testUrl = "www.somedomain.com";
+    globals.sandbox.stub(global.Services.io, "newURI").returns({host: testUrl});
+
+    const history = global.Cc["@mozilla.org/browser/nav-history-service;1"];
```

Comment:
cool trick. Any reason that `const history = Cc["@mozilla.org/browser/nav-history-service;1"].getService(Ci.nsINavHistoryService);` isn't done inside the contructor instead of at import time. Or even more lazily? Or does it not matter?

Reply:
Sorry for distracting with this kind of unrelated stuff 🙈

Reply:
Yeah, it should probably be lazy. Not sure why it was written to be access at the top level…